src/test/isolation2/input/alter_db_set_tablespace.source

```diff
@@ -0,0 +1,81 @@
+-- If a mirror checkpointer doesn't create checkpoint between making changes in
+-- a database and moving this database to another tablespace, then attempt to
+-- create checkpoint the next time should not result in the fsync error.
+
+
+-- start_matchsubs
+
+-- # Remove source file name and line
+-- m/^ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'\(null\)' \(.*\)/
+-- s/^ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'\(null\)' \(.*\)/ERROR:  failed to inject fault: ERROR:  fault not set, fault name:'(null)' (file:line)/
+
+-- end_matchsubs
+
+
+-- Setup
+!\retcode  mkdir -p @testtablespace@/adst_dest;
+-- Set fsync on since we need to test the fsync code logic.
+!\retcode gpconfig -c fsync -v on --skipvalidation;
+!\retcode gpstop -u;
+CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
+
+CREATE OR REPLACE FUNCTION mirror0() RETURNS gp_segment_configuration.dbid%type
+AS $$
+    SELECT dbid FROM gp_segment_configuration WHERE content = 0 AND role = 'm'
+$$ LANGUAGE SQL EXECUTE ON MASTER;
+
+-- Checkpointer must not start processing requests until database is moved
+-- to another tablespace
+SELECT gp_inject_fault('ckpt_loop_begin', 'infinite_loop', mirror0());
+
+-- Wait until checkpointer is ready to start processing requests
+SELECT gp_wait_until_triggered_fault('ckpt_loop_begin', 1, mirror0());
+
+-- Create the destination tablespace
+CREATE TABLESPACE adst_dest LOCATION '@testtablespace@/adst_dest';
+
+-- Create a database in the default tablespace
+CREATE DATABASE alter_db;
+
+-- Make a change in the database
+1:@db_name alter_db: CREATE TABLE t(i int) DISTRIBUTED RANDOMLY;
+1q:
+
+-- Prepare faults for waiting. The could_not_fsync_file fault has the
+-- infinite_loop type so that the checkpointer shutdown occurs after the test
+-- is completed
+SELECT gp_inject_fault('ckpt_loop_end', 'skip', mirror0());
+SELECT gp_inject_fault('could_not_fsync_file', 'infinite_loop', mirror0());
```

Comment:
Why do we need `skip` injection here? Skip makes possible to guard a code block with condition, but you don't use this possbility. Moreover, as far I understand, we never complete checkpoint (process never achieve `ckpt_loop_end`). Maybe use it as error marker without `could_not_fsync_file`?

Reply:
`skip` is required to prepare fault for using in `gp_wait_until_triggered_fault()`.This function doesn't work with faults which are not activated. `skip` is the most suitable way to activate this fault. Other ways (`sleep`, `fatal`,`panic`, `error`, `infinite_loop`, `suspend`, `segv`) do something redundant.